src/chart/generateCategoricalChart.tsx

```diff
@@ -174,14 +174,10 @@ const getDisplayedData = (
 
     return result;
   }, []);
-  if (itemsData && itemsData.length > 0) {
+  if (itemsData.length > 0) {
     return itemsData;
   }
 
-  if (item && item.props && item.props.data && item.props.data.length > 0) {
-    return item.props.data;
-  }
```

Comment:
why can this be deleted?

Reply:
item is an element of graphicalItems, and from the code alone, it's not clear how it is being used. If passing item is necessary, it can be passed as part of graphicalItems, so I removed it.

Reply:
Agreed but unfortunately not much else is clear by code alone either 😅. We don't have the luxury of making changes like this without knowing the complete set of implications, especially when refactoring. 

Not ideal, but it's what we've been dealt

Reply:
I believe I've checked everything overall, but I can revert it if you want. it doesn't seem that important.

The chart and each of its elements have the data props, but there are too many inconsistencies and ambiguities in the handling process. 😢😢😢

Reply:
Agree, it's hard to work with which is why we're trying to make these changes. Thanks for the effort, it's super appreciated. 

Let me pull it down later/tomorrow and test

Reply:
pulled it down, and yeah! this is totally fine to remove. I think the GitHub UI just makes it difficult to see how everything is connected when reviewing. Nice cleanup!